cli/types/cypress.d.ts

```diff
@@ -5131,6 +5131,21 @@ declare namespace Cypress {
     tag?: string
   }
 
+  interface DevServerOptions {
+    specs: Cypress.Cypress['spec'][]
+    config: {
+      supportFile: string
+      projectRoot: string
+      webpackDevServerPublicPathRoute: string
+    }
+  }
+
+  interface ResolvedDevServerConfig {
+    port: number;
+    // TODO: must improve this typing
+    server: any;
```

Comment:
This can be an HttpServer

Reply:
Where would this typing be defined? 
Should it be imported from http like its definition in Webpack-dev-server? 
This would make the `cypress.d.ts` change from ambient to module, which entails a lot of breaking change.

Should it be cloned? Copy/pasted?
I am looking onto better ways. 
Will report later.

Reply:
The http module does not declare global namespaces (better for isolation and worse for our case).
I had to do a few changes in how the file is structured, going from global references to an actual typings module (since valid since TS 2.8 I believe).
The way it is used is the same but the way it is written is a bit different.

What do you think?